apps/js_urls/serializer.py

```diff
@@ -0,0 +1,130 @@
+# Python Standard Library Imports
+import json
+from urllib.parse import urljoin
+
+# Django Imports
+from django.urls import get_resolver
+from django.urls.resolvers import (
+    RegexPattern,
+    RoutePattern,
+    URLPattern,
+    URLResolver,
+)
+
+# HTK Imports
+from htk.apps.js_urls.constants.regexp import (
+    URL_ARG_RE,
+    URL_KWARG_RE,
+    URL_OPTIONAL_CHAR_RE,
+    URL_OPTIONAL_GROUP_RE,
+    URL_PATH_RE,
+)
+from htk.apps.js_urls.utils import replace
+from htk.utils import htk_setting
+
+
+def get_urls_as_dict(resolver=None):
+    resolver = resolver or get_resolver()
+    return dict(_parse_resolver(resolver))
+
+
+def get_urls_as_json(resolver=None):
+    resolver = resolver or get_resolver()
+    return json.dumps(get_urls_as_dict(resolver))
+
+
+def _parse_resolver(
+    resolver, current_namespace=None, url_prefix=None, include_all=False
```

Comment:
Any reason to name the arg `current_namespace` as opposed to just `namespace`?

Reply:
This is a recursive function. And Django URL namespaces can be nested.